rules/fusesoc.bzl

```
@@ -36,15 +36,20 @@ def _fusesoc_build_impl(ctx):
         verilator_options = ctx.attr.verilator_options[BuildSettingInfo].value
         flags.append("--verilator_options={}".format(" ".join(verilator_options)))
 
+    cores_roots = {}
+    for c in ctx.files.cores:
+        if c.dirname not in cores_roots:
+            cores_roots[c.dirname] = True
+
```

nit: A `set` might be the data type that more directly expresses the intention here:
```suggestion
    cores_roots = set()
    for c in ctx.files.cores:
        cores_roots.add(c.dirname)
```

or
```python
    cores_roots = set([c.dirname for c in ctx.files.cores])
```

---

Huh... I was today years old when I learned that "set" was built into Python, but I don't think they exist in Starlark so I wouldn't expect this to work.

---

Oh yeah... It's Starlark. Doh, I forgot! I can't decide whether to be bothered by the "sort of Python" nature of this language. But in any case, withdrawn!